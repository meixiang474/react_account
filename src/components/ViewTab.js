import React, {memo, useState, useCallback} from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Ionicon from 'react-ionicons';
import {LIST_VIEW, CHART_VIEW} from '../util';

const ViewTab = memo(props => {
  const {
    activeTab,
    onTabChange
  } = props

  const [current, setCurrent] = useState(activeTab)

  const clickList = useCallback((e) => {
    e.preventDefault();
    setCurrent(LIST_VIEW);
    onTabChange(LIST_VIEW);
  }, [onTabChange])

  const clickChart = useCallback((e) => {
    e.preventDefault();
    setCurrent(CHART_VIEW);
    onTabChange(CHART_VIEW);
  }, [onTabChange])

  return (
    <ul className="nav nav-tabs nav-fill my-4">
      <li className="nav-item">
        <a 
          className={classnames('nav-link', {active: current === LIST_VIEW})} 
          href="#"
          onClick={(e) => clickList(e)}
        >
          <Ionicon
            className="rounded-circle mr-2"
            fontSize="25px"
            color="#007bff"
            icon="ios-paper"
          /> 
          列表模式
        </a>
      </li>
      <li className="nav-item">
        <a 
          className={classnames('nav-link', {active: current === CHART_VIEW})} 
          href="#"
          onClick={(e) => clickChart(e)}
        > 
          图表模式
        </a>
      </li>
    </ul>
  )
});

ViewTab.propTypes = {
  activeTab: PropTypes.string.isRequired,
  onTabChange: PropTypes.func.isRequired
}

ViewTab.defaultProps = {
  activeTab: LIST_VIEW,
  onTabChange: () => {}
}

export default ViewTab