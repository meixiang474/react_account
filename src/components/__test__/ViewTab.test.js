import React from 'react';
import {shallow} from 'enzyme';
import ViewTab from '../ViewTab';
import {LIST_VIEW, CHART_VIEW} from '../../util';

const props = {
  activeTab: LIST_VIEW,
  onTabChange: jest.fn()
}

const e = {
  preventDefault: jest.fn()
}

let wrapper;
describe('test ViewTab Component', () => {
  beforeEach(() => {
    wrapper = shallow(<ViewTab {...props}/>)
  })
  it('should render the component match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render tab active correct', () => {
    expect(wrapper.find('ul li').first().find('a').props().className).toEqual('nav-link active')
  })
  it('should trigger correct function callback', () => {
    const a = wrapper.find('ul li').first().find('a')
    a.simulate('click', e)
    expect(props.onTabChange).toHaveBeenCalledWith(LIST_VIEW)
  })
})