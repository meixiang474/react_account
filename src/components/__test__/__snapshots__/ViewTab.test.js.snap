// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ViewTab Component should render the component match snapshot 1`] = `ShallowWrapper {}`;
